### PR TITLE
Connect mongodb driver and maintain db connection in backend

### PR DESCRIPTION
**Pull request by ranjanshivangi**

### CODE REVIEW

File: src/services/user.service.js [hunk 1]
```diff
@@ -1,39 +1,33 @@
...
-export const getAllUsers = async () => {
-  const data = await User.find();
-  return data;
-};
-
-//create new user
 export const newUser = async (body) => {
-  const data = await User.create(body);
-  return data;
+  console.log(body);
```
Comment (pragyaquo) [inline src/services/user.service.js:6]: remove consoles.

File: src/services/user.service.js [hunk 2]
```diff
@@ -1,39 +1,33 @@
...
+  const userPresent = await User.findOne({ email: body.email })
+  if (userPresent == null) {
+    const salt = await bcrypt.genSalt(10);
+    body.password = await bcrypt.hash(body.password, salt);
+    console.log(body)
```
Comment (pragyaquo) [inline src/services/user.service.js:11]: if user changes password , where will the previous password be saved ..?

File: src/services/user.service.js [hunk 3]
```diff
@@ -1,39 +1,33 @@
...
-      new: true
+export const login = async (body) => {
+  const userPresent = await User.findOne({ email: body.email });
+  if (userPresent == null) {
+    throw new Error("User does not exist");
+  } else {
+    const validPassword = await bcrypt.compare(body.password, userPresent.password);
+    if (validPassword) {
+      let token = jwt.sign({ id: userPresent._id}, process.env.SECRET_KEY);
+      return { user: userPresent, token };
```
Comment (pragyaquo) [inline src/services/user.service.js:28]: should token be going in api response ?

File: src/services/user.service.js (hunk 1 above)
Comment (ranjanshivangi) [reply src/services/user.service.js:6]: ok

File: src/services/user.service.js (hunk 2 above)
Comment (ranjanshivangi) [reply src/services/user.service.js:11]: here we are not changing the password, we are just hashing it before saving it to database

**Pull request merged**